src/components/SelectionList/Search/TaskListItemRow.tsx

```diff
@@ -0,0 +1,260 @@
+import Badge from '@components/Badge';
+import Button from '@components/Button';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import type {TaskListItemType} from '@components/SelectionList/types';
+import TextWithTooltip from '@components/TextWithTooltip';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {callFunctionIfActionIsAllowed} from '@libs/actions/Session';
+import {completeTask} from '@libs/actions/Task';
+import DateUtils from '@libs/DateUtils';
+import Parser from '@libs/Parser';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ReportInfoCell from './ReportInfoCell';
+import UserInfoCell from './UserInfoCell';
+
+type TaskListItemRowProps = {
+    item: TaskListItemType;
+    showTooltip: boolean;
+    containerStyle?: StyleProp<ViewStyle>;
+};
+
+type CellProps = {
+    // eslint-disable-next-line react/no-unused-prop-types
+    showTooltip: boolean;
+    isLargeScreenWidth: boolean;
+};
+
+type TaskCellProps = {
+    taskItem: TaskListItemType;
+} & CellProps;
+
+function DateCell({taskItem, showTooltip, isLargeScreenWidth}: TaskCellProps) {
+    const styles = useThemeStyles();
+
+    const created = taskItem.created;
+    const date = DateUtils.formatWithUTCTimeZone(created, DateUtils.doesDateBelongToAPastYear(created) ? CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT : CONST.DATE.MONTH_DAY_ABBR_FORMAT);
+
+    return (
+        <TextWithTooltip
+            text={date}
+            shouldShowTooltip={showTooltip}
+            style={[styles.lineHeightLarge, styles.pre, styles.justifyContentCenter, isLargeScreenWidth ? undefined : [styles.textMicro, styles.textSupporting]]}
+        />
+    );
+}
+
+function TitleCell({taskItem, showTooltip, isLargeScreenWidth}: TaskCellProps) {
+    const styles = useThemeStyles();
+    const taskTitle = Parser.replace(Parser.htmlToText(taskItem.reportName));
+
+    return (
+        <TextWithTooltip
+            text={taskTitle}
+            shouldShowTooltip={showTooltip}
+            style={[isLargeScreenWidth ? styles.lineHeightLarge : styles.lh20, styles.pre, styles.justifyContentCenter]}
+        />
+    );
+}
+
+function DescriptionCell({taskItem, showTooltip, isLargeScreenWidth}: TaskCellProps) {
+    const styles = useThemeStyles();
+    const taskDescription = Parser.replace(Parser.htmlToText(taskItem.description)).replaceAll('<br />', ' ');
+
+    return (
+        <TextWithTooltip
+            shouldShowTooltip={showTooltip}
+            text={taskDescription}
+            style={[styles.lineHeightLarge, styles.pre, styles.justifyContentCenter, isLargeScreenWidth ? undefined : [styles.textMicro, styles.textSupporting]]}
+        />
+    );
+}
```

Comment:
Same for these. They are just texts in the end. Can we instead add these fields to the item we're rendering and then these components are just templates to render data, without any business logic?